### ISSUE TITLE
Figuring out tiles

### ISSUE THREAD

**Issue by checkas** [SourMesen/Mesen#848]
Please pardon my ignorance here I'm trying to find out if there is an easier way of reading the tile information for games whenever building an HD pack? When looking at a title sequence for a game I have noticed that the elements that might create a logo are spread out throughout different tiles with some repeating, which I knew that some elements would be reused. Is there a practical way to determine this so that whenever I'm editing in Photoshop I'll know what to use or how to map it or where the mapping might be? In need help you all can provide would be appreciated as I'm trying to learn more about this in the best manner possible.

**Comment by mkwong98** [comment]
Play the game until the point where the object can be seen on the screen (next frame to be exact). Pause the emulator and select "PPU Viewer" under "Debug" menu. For background tiles, under the "Nametable Viewer", there will be 4 screen showing all the background tiles in the nametable. Hold the "Shift" key and right click a screen and choose "Copy Nametable (HD Pack Format)".  This will copy the tiles into the clipboard. Then you can either:
1. Paste the data in a text file, save it with csv extension and open it in Excel. You can now use Excel formula or sort function to look for duplicates. 
2. Paste the data in my HD pack editor and there is an option to show duplicate tiles with no assigned condition. https://drive.google.com/open?id=1BDpCniYpbYvRZS-JQd59awfpy3SIVXjQ

**Comment by checkas** [comment]
So there isnt any way to take the jumbled logo and get it to where it's normal without doing it by hand?
-------- Original message --------From: mkwong98 <notifications@github.com> Date: 6/10/20  1:07 AM  (GMT-06:00) To: SourMesen/Mesen <Mesen@noreply.github.com> Cc: checkas <justin@justintoney.com>, Author <author@noreply.github.com> Subject: Re: [SourMesen/Mesen] Figuring out tiles (#848) 
Play the game until the point where the object can be seen on the screen (next frame to be exact). Pause the emulator and select "PPU Viewer" under "Debug" menu. For background tiles, under the "Nametable Viewer", there will be 4 screen showing all the background tiles in the nametable. Hold the "Shift" key and right click a screen and choose "Copy Nametable (HD Pack Format)".  This will copy the tiles into the clipboard. Then you can either:

Paste the data in a text file, save it with csv extension and open it in Excel. You can now use Excel formula or sort function to look for duplicates.
Paste the data in my HD pack editor and there is an option to show duplicate tiles with no assigned condition. https://drive.google.com/open?id=1BDpCniYpbYvRZS-JQd59awfpy3SIVXjQ


—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/SourMesen/Mesen/issues/848#issuecomment-641745783",
"url": "https://github.com/SourMesen/Mesen/issues/848#issuecomment-641745783",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by SourMesen** [comment]
Yes, unfortunately there is no way to do this automatically. Mesen records the tiles as they are stored in the .nes file.  Originally, I tried to make the recorder "smarter" and to try and arrange the tiles like how they appeared on the screen, but it didn't work very well and was generally worse than just using the same order as the .nes file, for most scenarios.
